### PR TITLE
AV-184601: Fix for HTTPRule with empty LBAlgorithm

### PR DESCRIPTION
**Pull request by swathinsankaran**
Cherry-pick of #1225

### CODE REVIEW

**Comment by swathinsankaran** [comment]
build ako

**Review by akshayhavile — APPROVED** [review]

**Comment by swathinsankaran** [comment]
Please merge this once the PR #1228 is merged.

**Pull request merged**
